### ISSUE TITLE
FilesModel::findByPid() funktioniert nicht mit UUIDs

### ISSUE THREAD

**Issue by websolutions-hamburg** [contao/core-bundle#925]
Hallo,

die Funktion `\FilesModel::findByPid()` scheint noch nicht für UUIDs vorbereitet zu sein.

Funktionen wie `\FilesModel::findById()` bzw. deren Weiterleitung an die Funktion `\FilesModel::findByUuid()` haben eine spezielle Abfrage für UUIDs und eine passende Format Umwandlung mit Hilfe von `\StringUtil::uuidToBin($strUuid)`.

Bei der Funktion `\FilesModel::findByPid()` werden die UUIDs jedoch nicht in ein Binary Format umgewandelt, sondern direkt an den SQL Select Befehl weitergegeben. Dadurch funktioniert dieser dann nicht richtig und liefert keine Ergebnis.

**Comment by fritzmg** [comment]
Would make sense, but what exactly is your use case though? Are you storing folder UUIDs as strings somewhere?

**Comment by websolutions-hamburg** [comment]
Not storing, but I select a folder with

```
'inputType' => 'fileTree',
'eval' => array(
    'fieldType' => 'radio',
    'mandatory' => false,
    'files' => false,
    'filesOnly' => false
)
```

and get the UUID from the folder. Later I will get all files from this folder. Maybe I can use `\FilesModel::findMultipleFilesByFolder()`. But for this I need 2 SQL select calls. The first to get the folder path from the UUID and the second to get the files.
With `\FilesModel::findByPid()` it will be only one select call. Or do I overlook something?

**Comment by fritzmg** [comment]
Post the complete DCA definition of your field.

**Comment by websolutions-hamburg** [comment]
I use the extension [contao-rocksolid-custom-elements](https://github.com/madeyourday/contao-rocksolid-custom-elements) with this DCA:

```
'label' => array(
    'Hintergrundbilder',
    'Lädt die Hintergrundbilder'),
'types' => array('content'),
'contentCategory' => 'media',

'fields' => array(
    'images_folder' => array(
        'label' => array(
            'Ordner',
            'Wählen Sie den Ordner.'),
        'inputType' => 'fileTree',
        'eval' => array(
            'fieldType' => 'radio',
            'mandatory' => false,
            'files' => false,
            'filesOnly' => false
        )
    )
)
```

And in the config file of this custom element I get the UUID of the folder with the `$this->images_folder` variable.

**Comment by fritzmg** [comment]
> I use the extension contao-rocksolid-custom-elements with this DCA:

Ah, that's the problem then. Usually if you use the `fileTree` widget, your SQL storage field for storage would be `blob` or `binary(16)`. However, with RSCE, the value is probably transformed by the extension from binary to string before being saved in `rsce_data`.

Nevertheless, as I said it would make sense to introduce a conversion like in `findByUuid` as well :) 👍

**Comment by leofeyer** [comment]
A method `findByUuid()` exists. I think @websolutions-hamburg needs a `findByPid()` with a built-in `if (Validator::isUuid(` check.

**Comment by fritzmg** [comment]
Yes, that's what I meant.

**Comment by websolutions-hamburg** [comment]
> ... with RSCE, the value is probably transformed by the extension from binary to string before being saved in rsce_data.

Ah ok. Good to know ;-)
Thanks for this information!

@leofeyer yes, that's right.

**Comment by leofeyer** [comment]
Hm, I just wanted to implement the method, however, it does not really make sense, does it? `tl_files.pid` is always a UUID, so what exactly do you want the method to do?

**Comment by ausi** [comment]
> so what exactly do you want the method to do?

Convert string UUIDs that get passed into the method to binary UUIDs?

**Comment by leofeyer** [comment]
Like this?

```php
public static function findByPid($intPid, array $arrOptions=array())
{
	$t = static::$strTable;

	// Convert UUIDs to binary
	if (\Validator::isStringUuid($intPid))
	{
		$intPid = \StringUtil::uuidToBin($intPid);
	}

	return static::findBy(array("$t.uuid=UNHEX(?)"), bin2hex($intPid), $arrOptions);
}
```

**Comment by ausi** [comment]
Yes 👍

**Comment by leofeyer** [comment]
Implemented in 9f5adad278a3ccd251a8ccab220a0326a202feaa.
